### PR TITLE
redirect to xlsx export when there are too many rows for xls

### PR DESCRIPTION
**Pull request by nickpell**
http://manage.dimagi.com/default.asp?189139

@czue @esoergel cc: @kaapstorm 

It would be nice if there was an alert to the user that the export format was changed (it's not clear what page would even be appropriate for that warning since not all users can see the data source config editor), but this covers a good 80% of the use case with 20% of the effort.

### CODE REVIEW

File: corehq/apps/userreports/views.py [hunk 1]
```diff
@@ -699,6 +699,18 @@ def export_data_source(request, domain, config_id):
     for sql_filter in params.sql_filters:
         q = q.filter(sql_filter)
 
+    # xls format has limit of 65536 rows
+    # First row is taken up by headers
+    if params.format == 'xls' and q.count() >= 65535:
+        keyword_params = dict(**request.GET)
+        keyword_params.update(format='xlsx')
```
Comment (czue) [inline corehq/apps/userreports/views.py]: couchexport has constants for these that are used in most of the code over the strings (even though that's a little silly)

**Comment by czue** [comment]
i don't love this solution but agree that it's probably an okay compromise

**Comment by kaapstorm** [comment]
I guess this might be a question for Product, but what are the benefits of supporting both XLS and XLSX? Are there users out there who don't have an Office product less than 15 years old? Would it clean things up just to use XLSX exclusively?

File: corehq/apps/userreports/views.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/userreports/views.py]: oh good point: https://github.com/dimagi/commcare-hq/pull/9536

**Comment by nickpell** [comment]
> I guess this might be a question for Product, but what are the benefits of supporting both XLS and XLSX? Are there users out there who don't have an Office product less than 15 years old? Would it clean things up just to use XLSX exclusively?

@dimagi/product

**Comment by amsagoff** [comment]
Oh man, I wish I knew the answer to that question.

Would it make sense to change the format to csv rather than xlsx? That way,
if people are actually using xls, they can still open the file in excel.

Also open to removing the xls option from new custom exports as well.

**Comment by nickpell** [comment]
@amsagoff That makes a lot of sense.  Updated the format to CSV here: https://github.com/dimagi/commcare-hq/pull/9536

**Pull request merged**